# Radicle Grants Program (RGP) - Committee Compensation

The analysis below looks at the last 6 months of RAD prices, denominated in USD terms.

The main goal of the analysis is to gauge a very conservative estimate of the amount of RAD needed in the RGP's multi-sig wallet in order to pay committee members. This is to ensure that we have enough RAD to pay committee members in meaningful equity without running out of RAD due to fluctuations in conversion rates. 

There are a few charts to show the price over time, but these are not important to the outcome of the analysis.

Source: https://www.coingecko.com/en/coins/radicle/historical_data/usd?end_date=2021-11-24&start_date=2021-05-29#panel

In [22]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

In [24]:
time_series_df = pd.read_csv('rad-usd-max.csv')
time_series_df

url = "https://www.coingecko.com/price_charts/export/14013/usd.csv"
time_series_df = pd.read_csv(url)

snapped_at      price    market_cap  total_volume
0    2021-02-25 00:00:00 UTC  11.646011  0.000000e+00  1.107455e+07
1    2021-02-26 00:00:00 UTC  11.646011  0.000000e+00  1.107455e+07
2    2021-02-27 00:00:00 UTC  11.724064  0.000000e+00  1.617266e+07
3    2021-02-28 00:00:00 UTC  15.625533  0.000000e+00  2.365203e+07
4    2021-03-01 00:00:00 UTC  13.411671  0.000000e+00  2.796986e+06
..                       ...        ...           ...           ...
268  2021-11-20 00:00:00 UTC  13.121472  2.773117e+08  1.781942e+07
269  2021-11-21 00:00:00 UTC  13.093094  2.768384e+08  1.438152e+07
270  2021-11-22 00:00:00 UTC  13.429398  2.839268e+08  3.499454e+07
271  2021-11-23 00:00:00 UTC  12.470071  2.645682e+08  1.735316e+07
272  2021-11-24 00:00:00 UTC  12.334478  2.608209e+08  2.007043e+07

[273 rows x 4 columns]

In [25]:
price_time_series_df = time_series_df['price']
price_time_series_df

0      11.646011
1      11.646011
2      11.724064
3      15.625533
4      13.411671
         ...    
268    13.121472
269    13.093094
270    13.429398
271    12.470071
272    12.334478
Name: price, Length: 273, dtype: float64

## Average Price

In [27]:
average_price = price_time_series_df.mean()
average_price

10.803550057143621

The average price of RAD was $10.80 (USD) in the last 6 months (180 days).

## Standard Deviation

In [29]:
standard_deviation = price_time_series_df.std()
standard_deviation

4.682272817590489

The standard deviation in price was +/- $4.68 (USD).

## Outcome/Takeaway

The main goal here is to ensure we have enough RAD to pay RGP Committee members in RAD, even with flucuations in the RAD:USD conversion rate.

Let's start with some key points:
* We have 6 committee members
* A high estimate of time worked would be 50 hours per month, over 6 months. This totals to 1,800 hours.
* At 150 (USD) per hour, this would total 270,000 (USD) worth of RAD for the entire 6 months.

In [30]:
committee_members = 6
monthly_hours = 50
grants_program_lengths_in_months = 6
total_hours_worked = committee_members * monthly_hours * grants_program_lengths_in_months

hourly_compensation_usd = 150
total_rad_compensation_usd = total_hours_worked * 150
total_rad_compensation_usd

270000

I propose that we use the following formula to determine the amount of RAD needed to be transferred from the Radicle Treasury to the RGP's multi-sig wallet:

`RAD to Transfer = $270,000 / (AVG Price over 6 months - STD of price)`


In [31]:
rad_to_transfer = total_rad_compensation_usd / (average_price - standard_deviation)
rad_to_transfer

44108.441659099015

Or in plain English: we should transfer 44,108 RAD from the Radicle Treasury to the RGP's multi-sig wallet.

Any unused funds at the end of the 6 months would be transferred back to the Radicle Treasury.